source/common/http/http2/codec_impl.cc

```diff
@@ -146,7 +146,8 @@ void ConnectionImpl::StreamImpl::encodeHeadersBase(const std::vector<nghttp2_nv>
 
   local_end_stream_ = end_stream;
   submitHeaders(final_headers, end_stream ? nullptr : &provider);
-  parent_.sendPendingFrames();
+  auto status = parent_.sendPendingFrames();
+  RELEASE_ASSERT(status.ok(), "sendPendingFrames() failure in non dispatching context");
```

Comment:
https://github.com/envoyproxy/envoy/pull/11503
Ideally would be replaced with ENVOY_BUGs

Reply:
See https://github.com/envoyproxy/envoy-setec/pull/181 for other context on this. This whole situation is a total mess on many accounts. I actually think with the current code the only sane thing to do is to crash here, as in the current code this would be an uncaught exception. I think using ENVOY_BUG here would lead to undefined behavior.

As I mention in the other PR/issue, I think the only sane we to handle errors in non-dispatch context is to latch them as mentioned, and then handle them on the next dispatch. Another option (depending on context) would be to close the underlying connection and let the close event do its job. 

I think the question is whether we want to try to fix ^ in this PR, or do it iteratively. I would be in favor of doing it iteratively.

Reply:
I think iteratively is fine, we should add some clear TODOs explaining this.

Reply:
I have looked through (I think) all call stacks to make sure that this RELEASE_ASSERT did not modify the existing behavior. It should just replace abnormal termination due to unhandled exception. Yes all this will be cleaned up as part of hardening against untrusted upstreams. 
Current PoR is to latch the error for later handling. But this is a much bigger change.
Also this change will be flag protected, so if someone is hitting any of these RELEASE_ASSERTS they can go flip to old codecs with exceptions until the problem is resolved.

Reply:
> Yes all this will be cleaned up as part of hardening against untrusted upstreams.

Pedantically (and I know you know this), this is not just about untrusted upstreams. In the current code it's possible (I think) to hit flood exceptions in the out of dispatch encode cases. We need to protect against this also.

But yes I think we are all on the same page about latching.

Reply:
I have not problem replacing RELEASE_ASSERT with that new macro we have added recently to do logging and stat increment. My gut feeling is that it might not be much of an improvement, given that it is very likely something else will crash elsewhere very soon as we are proceeding with processing data where before we would not. It is possible however that in some cases it would not crash, and it is worth reducing the likelihood of the worst outcome.

Reply:
done